students/srepking/Lesson04/mailroom2.py

```diff
@@ -0,0 +1,122 @@
+def thank_you():
+    print('\n''Ok, you chose 1, lets see what we can do.')
+    while True:
+        name = input('\n''Type the full name to add entry, '
+                     '\'list\' to see list of names, or \'e\' to exit: > ')
+        # See if they want a list of names.
+        if name != 'list':
+            break
+        for x in donors:
+            print(x)
+    # Add Name to list if not in list
+    if name not in donors and name != 'e':
+        print('Name is not there, added to the list')
+        donors[name] = []
+    if name != 'e':
+        amount = input('\n''What is the donation amount? or \'e\' to exit >')
+        if amount != 'e':
+            new_amount = [float(amount)]
+            old_amount = donors[name]
+            donors[name] = old_amount + new_amount
+            print('Thank you so much for the generous gift of ${0:.2f}, {1}!'
+                  .format(float(amount), name))
+
+
+def report():
+    print('\n''A summary of your donors donations:')
+    # Create a new dictionary with Total, number of donations,
+    # and average donation amount
+    donors_f = {}
+    for name, donations in donors.items():
+        # Calculate total donation amount
+        total = 0
+        count = 0
+        average = 0
+        i = 0
+        for i in donations:
+            total = total + i
+        # Calculate total number of donations, not counting $0
+        for i in donations:
+            if i != 0:
+                count = count + 1
```

Comment:
use `sum` for total and `len` for count logic

Reply:
I implemented the sum() function. However, I don't think the len() function will work for what I am doing. I am not counting donations of $0 as donations. If I use len(), I have to assume all numbers in my list are donations, and I thought that a donation of $0 should be disregarded. Or  is there another way to use len() I am overlooking? Thank you!

Reply:
I see the new Pull Request and will take a look at the latest code and look into more details for this part, I will provide recommendation there.